src/ast/ast_validations.c

```diff
@@ -1705,3 +1688,39 @@ AST_Validation AST_Validate(RedisModuleCtx *ctx, const cypher_parse_result_t *re
 	return res;
 }
 
+AST_Validation AST_Validate_QueryParams(RedisModuleCtx *ctx, const cypher_parse_result_t *result) {
```

Comment:
I think that the logic required by this function and `AST_Validate_Query` should be in a shared function, rather than being copied. That said, this function should only contain the checks that are sensible for parameter checking - for example, can we have a NULL root when parsing parameters? Can we encounter a non-statement root?